site/en/docs/web-platform/navigation-api/index.md

```diff
@@ -96,116 +92,96 @@ links.forEach(link => link.addEventListener('click', updatePage));
-Your site might choose to do this in response to a user action, or perhaps if the loaded URL is invalid and should never appear in the user's history.
-For example:
-
-```js
-appHistory.addEventListener('navigateerror', event => {
-  const attemptedLoadURL = location.href;
-  const previousURL = appHistory.transition.from;
-  console.warn('Failed to load', attemptedLoadURL, 'returning to', previousURL);
-  appHistory.transition.rollback();
-});
-```
-
-Like many of the features in the App History API, this call to `rollback()` is a semantic action that implies meaning to the browser.
-If a clicked `<a href>` fails to load and rolls back, the URL won't appear in the user's session history (for Forward or Back navigation).
-
-{% Aside %}
-
-Although calling `rollback()` on error is useful in some situations, you should avoid it in the general case.
-You're more likely to show an error message, perhaps so the user can try a reload to get their content.
-
-{% endAside %}
-
 ### Abort Signals
 
 Since you're able to do asynchronous work while preparing a new page, it's possible that the transition your code is handling (to load a specific URL or state) might get preempted, or considered out-of-date.
-This might happen because the `rollback()` method was invoked, the user just clicked on another link, or your code itself throws an error.
+This might happen because the user just clicked on another link, or your code itself throws an error.
 
-To deal with any of these possibilities, the event passed to the "navigate" handler contains a property called `signal`, which is an instance of `AbortSignal`.
+To deal with any of these possibilities, the event passed to the "navigate" listener contains a `signal` property, which is an `AbortSignal`.
 For more information see [Abortable fetch][abortable-fetch].
 The short version is it basically provides an object that fires an event when you should stop your work.
 Notably, you can pass an `AbortSignal` to any calls you make to `fetch()`, which will cancel in-flight network requests if the navigation is preempted.
 This will both save the user's bandwidth, and reject the `Promise` returned by `fetch()`, preventing any following code from e.g., updating the DOM to show a now invalid page navigation.
 
-For a concrete example, you might set up loading a page of cat memes with a `fetch()` call in your handler.
+For a concrete example, you might set up loading a page of cat memes with a `fetch()` call in your listener.
 By passing the `signal` to it, the fetch will be cancelled if the user decides to instead load a different page on your site before the `fetch` completes.
 Take a look:
 
 ```js
-appHistory.addEventListener('navigate', event => {
-  if (isCatsUrl(event.destination.url)) {
+navigation.addEventListener('navigate', navigateEvent => {
+  if (isCatsUrl(navigateEvent.destination.url)) {
     const method = async () => {
```

Comment:
Preexisting issue... this is more of a function than a method... maybe name it something like processNavigation.